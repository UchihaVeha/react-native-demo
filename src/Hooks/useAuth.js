import { useState, useEffect } from 'react';
import auth from '@react-native-firebase/auth';

export default function useAuth() {
  const [authState, setState] = useState({
    isLoading: true,
    isLoggedIn: false,
    error: null
  });

  useEffect(() => {
    async function bootstrap() {
      try {
        await auth().signInAnonymously();
        setState(() => ({
          isLoading: false,
          isLoggedIn: true,
          error: null
        }));
      } catch (e) {
        switch (e.code) {
          case 'auth/operation-not-allowed':
            setState(() => ({
              isLoading: false,
              isLoggedIn: false,
              error: 'Enable anonymous in your firebase console.'
            }));
            break;
          default:
            setState(() => ({
              isLoading: false,
              isLoggedIn: false,
              error: 'Sorry, some error happened'
            }));
            break;
        }
      }
    }
    bootstrap();
  }, []);

  return authState;
}
