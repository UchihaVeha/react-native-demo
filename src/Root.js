import React from 'react';
import { StyleSheet, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import {
  Provider as PaperProvider,
  ActivityIndicator
} from 'react-native-paper';
import HomeScreen from './Screens/HomeScreen';
import AddCompanyScreen from './Screens/AddCompanyScreen';
import useAuth from './Hooks/useAuth';

const Stack = createStackNavigator();

export default function App() {
  const { isLoading, isLoggedIn, error } = useAuth();
  // TODO handle error
  return (
    <PaperProvider testID="paperProvider">
      <NavigationContainer>
        {isLoading && (
          <View style={styles.container}>
            <ActivityIndicator />
          </View>
        )}
        {!isLoading && isLoggedIn && (
          <Stack.Navigator>
            <Stack.Screen name="Companies" component={HomeScreen} />
            <Stack.Screen name="AddCompany" component={AddCompanyScreen} />
          </Stack.Navigator>
        )}
      </NavigationContainer>
    </PaperProvider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  }
});
