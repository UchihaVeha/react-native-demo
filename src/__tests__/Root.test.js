import 'react-native';
import React from 'react';
import { render, shallow } from 'react-native-testing-library';
import Root from '../Root';
import useAuth from '../Hooks/useAuth';

jest.mock('../Hooks/useAuth');

it('should show indicator loading', async () => {
  useAuth.mockReturnValue({ isLoading: true, isLoggedIn: false });
  const { getByTestId } = render(<Root />);
  const { output: paperProvider } = shallow(getByTestId('paperProvider'));
  expect(paperProvider).toMatchSnapshot();
});

it('should show screens', async () => {
  useAuth.mockReturnValue({ isLoading: false, isLoggedIn: true });
  const { getByTestId } = render(<Root />);
  const { output: paperProvider } = shallow(getByTestId('paperProvider'));
  expect(paperProvider).toMatchSnapshot();
});
