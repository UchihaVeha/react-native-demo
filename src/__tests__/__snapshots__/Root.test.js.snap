// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show indicator loading 1`] = `
<Portal.Host>
  <Context.Provider
    value={
      Object {
        "icon": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "animation": Object {
            "scale": 1,
          },
          "colors": Object {
            "accent": "#03dac4",
            "backdrop": "rgba(0, 0, 0, 0.5)",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "notification": "#f50057",
            "onBackground": "#000000",
            "onSurface": "#000000",
            "placeholder": "rgba(0, 0, 0, 0.54)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": Object {
              "fontFamily": "System",
              "fontWeight": "300",
            },
            "medium": Object {
              "fontFamily": "System",
              "fontWeight": "500",
            },
            "regular": Object {
              "fontFamily": "System",
              "fontWeight": "400",
            },
            "thin": Object {
              "fontFamily": "System",
              "fontWeight": "100",
            },
          },
          "roundness": 4,
        }
      }
    >
      <ForwardRef(NavigationContainer)>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <ForwardRef />
        </View>
      </ForwardRef(NavigationContainer)>
    </ThemeProvider>
  </Context.Provider>
</Portal.Host>
`;

exports[`should show screens 1`] = `
<Portal.Host>
  <Context.Provider
    value={
      Object {
        "icon": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "animation": Object {
            "scale": 1,
          },
          "colors": Object {
            "accent": "#03dac4",
            "backdrop": "rgba(0, 0, 0, 0.5)",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "notification": "#f50057",
            "onBackground": "#000000",
            "onSurface": "#000000",
            "placeholder": "rgba(0, 0, 0, 0.54)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": Object {
              "fontFamily": "System",
              "fontWeight": "300",
            },
            "medium": Object {
              "fontFamily": "System",
              "fontWeight": "500",
            },
            "regular": Object {
              "fontFamily": "System",
              "fontWeight": "400",
            },
            "thin": Object {
              "fontFamily": "System",
              "fontWeight": "100",
            },
          },
          "roundness": 4,
        }
      }
    >
      <ForwardRef(NavigationContainer)>
        <StackNavigator>
          <Screen
            component={[Function]}
            name="Companies"
          />
          <Screen
            component={[Function]}
            name="AddCompany"
          />
        </StackNavigator>
      </ForwardRef(NavigationContainer)>
    </ThemeProvider>
  </Context.Provider>
</Portal.Host>
`;
